tests/unit/test_artifactory_path.py

```diff
@@ -967,5 +971,190 @@ def test_from_aql_file(self):
         assert artifact.root == "/reponame/"
 
 
+class TestArtifactoryPathGetAll(unittest.TestCase):
+    # TODO: test repositories and permissions
+    def setUp(self):
+        self.arti = ArtifactoryPath("http://b.com/artifactory")
+
+    def test_get_all_users(self):
+        url = "/api/security/users"
+        request_url = self.arti.drive + url
+        key = "name"
+        cls = User
+        users = [
+            {
+                "name": "user_1",
+                "uri": "http://b.com/artifactory/api/security/users/user_1",
+                "realm": "internal",
+            },
+            {
+                "name": "user_2",
+                "uri": "http://b.com/artifactory/api/security/users/user_2",
+                "realm": "internal",
+            },
+        ]
+        user_1 = {"name": "user_1", "email": "user1@example.com"}
+        user_2 = {"name": "user_2", "email": "user2@example.com"}
+
+        # lazy = True
+        with responses.RequestsMock() as rsps:
+            rsps.add(responses.GET, request_url, json=users, status=200)
+            results = self.arti._get_all(lazy=True, url=url, key=key, cls=cls)
+
+            map(self.assertIsInstance, zip(results, repeat(cls)))
```

Comment:
Does python3 call map right away? I thought it's a lazy function and you receive a map object

```python
>>> map(print, ['1', '2'])
<map object at 0x7f8811c92070>
>>> list(map(print, ['1', '2']))
1
2
[None, None]
>>> 
```

The one-liner is kinda tricky, let's use `for user in results` instead

Reply:
Thank you for pointing this. This issue is fixed